lib/matplotlib/backends/backend_qt.py

```diff
@@ -172,6 +174,9 @@ def _may_clear_sock():
         return sn  # Actually keep the notifier alive.
 
     def handle_sigint():
+        # Ensure notifier does not outlive the event loop shutting down:
+        sn.deleteLater()
```

Comment:
Should you also set `sn = None` to cleanup the Python side?

Reply:
Whilst it shouldn't be necessary because the Python objects all go out of scope soon enough, I think it's a good idea - better that incidental access to the Qt object later if there are bugs result in a `TypeError` than a SIGABRT and python exiting ungracefully, which is what would happen on trying to access a deleted C++ Qt object.